sw/device/tests/sim_dv/keymgr_key_derivation.c

```diff
@@ -0,0 +1,254 @@
+}
+
+static void init_flash(void) {
+  // setup default access for data partition
+  flash_default_region_access(/*rd_en=*/true, /*prog_en=*/true,
+                              /*erase_en=*/true);
+  // Initialize flash secrets.
+  write_info_page(kFlashInfoPageIdCreatorSecret, kCreatorSecret,
+                  ARRAYSIZE(kCreatorSecret));
+  write_info_page(kFlashInfoPageIdOwnerSecret, kOwnerSecret,
+                  ARRAYSIZE(kOwnerSecret));
+}
+
+/**
+ * Busy-wait until the DAI is done with whatever operation it is doing.
+ */
+static dif_otp_ctrl_t otp;
+
+/**
+ * Lock otp secret 2 partition so that on reboot flash seeds can be
+ * automatically loaded
+ */
+static void lock_otp_secret_partition2(void) {
+  // initialize otp
+  CHECK_DIF_OK(dif_otp_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp));
+
+  otp_ctrl_testutils_lock_partition(&otp, kDifOtpCtrlPartitionSecret2, 0);
+}
+
+/** Place kmac into sideload mode for correct keymgr operation */
+static void init_kmac_for_keymgr(void) {
+  dif_kmac_t kmac;
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Configure KMAC hardware using software entropy.
+  dif_kmac_config_t config = (dif_kmac_config_t){
+      .sideload = true,
+  };
+  CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
+
+  for (uint8_t i = 0; i < KMAC_PREFIX_SIZE; ++i) {
+    mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR),
+                        KMAC_PREFIX_0_REG_OFFSET + i * 4, kKmacPrefix[i]);
+  }
+}
+
+void keymgr_advance_state(const dif_keymgr_t *keymgr,
+                          const dif_keymgr_state_params_t *params) {
+  CHECK_DIF_OK(dif_keymgr_advance_state(keymgr, params));
+
+  while (true) {
+    dif_keymgr_status_codes_t status;
+
+    CHECK_DIF_OK(dif_keymgr_get_status_codes(keymgr, &status));
+    if (status == 0) {
+      LOG_INFO("Advancing to next state");
+    } else if (status == kDifKeymgrStatusCodeIdle) {
+      break;
+    } else {
+      LOG_ERROR("Unexpected status: %0x", status);
+      break;
+    }
+  }
+}
+
+void keymgr_check_state(const dif_keymgr_t *keymgr,
+                        const dif_keymgr_state_t exp_state) {
+  dif_keymgr_state_t act_state;
+  CHECK_DIF_OK(dif_keymgr_get_state(keymgr, &act_state));
+  if (act_state != exp_state) {
+    LOG_INFO("Keymgr in unexpected state: %0x, expected to be %0x", act_state,
+             exp_state);
+  }
+}
+
+bool test_main(void) {
+  dif_rstmgr_t rstmgr;
+  dif_rstmgr_reset_info_bitfield_t info;
+
+  // Initialize pwrmgr
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &info));
+
+  // POR reset
+  if (info == kDifRstmgrResetInfoPor) {
+    LOG_INFO("Powered up for the first time, program flash");
+
+    // Initialize flash
+    init_flash();
+
+    // Lock otp secret partition
+    lock_otp_secret_partition2();
```

Comment:
You probably forgot to replace this with `otp_ctrl_testutils_lock_partition()`.

Reply:
Ohh I see. you are invoking this in this static method. Maybe just inline `lock_otp_secret_partition2()` here directly. Fine doing this later also.

Reply:
Updated in the last push.